### PR TITLE
Read simple data cases automatically

### PR DESCRIPTION
**Pull request by saroad2**
### Description

When testing black formatting, the simple use cases where defined hard-coded in a list named `SIMPLE_CASES` in *test_format*.
This is a shame since many times people add a new data case and forget to add it to this list and then it does not run!
Now, one can read all the available cases in a given directory using a method named `all_data_cases`. This method searches for all the cases in a subdirectory of *data* and return them as names.

### Checklist - did you ...

- [X] Add a CHANGELOG entry if necessary?
- [X] Add / update tests if necessary?
- [X] Add new / update outdated documentation?

### Moving forward
If you approve this PR of simple cases, we can do the same for the other hard coded cases lists in *test_format.py*
Let me know if you want me to change those too.

### CODE REVIEW

File: tests/test_black.py
Comment (felix-hilden) [inline tests/test_black.py:1171]:
```suggestion
            p = THIS_DIR / "data" / "simple_cases" / "collections.py"
```
I know this is not your doing, but I'm pretty sure we don't need a double conversion if `THIS_DIR` is a path already 😅

**Review by felix-hilden — COMMENTED** [review]
Hi! Thanks for the PR. I like the idea, since it's easier to just put a file in a folder rather than having to edit a list somewhere else. We could definitely do this for all the test categories. I'd be okay with doing it in a future PR, but thought I'd mention it already.

EDIT: Unless we have the same file in multiple locations!

The changes themselves look alright to me. And to the best of my abilities I tried to verify that the correct files were indeed moved 😄 Nicely done!

**Comment by saroad2** [comment]
Hey @felix-hilden and @JelleZijlstra , thank you for the review!

Once this PR is approved, I will do the same for the other test cases.

**Review by felix-hilden — APPROVED** [review]
Yep, this looks fine! And thanks for considering taking on the whole task 🙏

**Comment by saroad2** [comment]
@felix-hilden, would you like me to make the change for all cases in this PR, or in a separate one?
If the latter, Please merge this PR and I will create a new one.

**Comment by felix-hilden** [comment]
A separate PR works just fine! But we like to have at least another maintainer have a look at things before merging so that we don't miss anything. Please stand by 😄

**Review by JelleZijlstra — APPROVED** [review]

File: CHANGES.md
Comment (ichard26) [inline CHANGES.md:70]: Do we need this? it's not user facing ...

**Review by ichard26 — COMMENTED** [review]
Would this kind of API for read_data be more developer-friendly?
```python
source, expected = read_data("simple_cases/expression.py")
```
vs what is currently proposed ...
```python
source, expected = read_data("expression.py", cases_dir="simple_cases")
```
Anyway, this PR looks great to me regardless.

**Comment by felix-hilden** [comment]
Certainly works too! If I had to come up with one, and given that soon we won't have any files in the top directory, we could have:

```python
def read(subdir: str, file: str):
    data_file = (TOP / subdir / file).with_suffix(".py")
```

or whatever. That would work around hypothetical slash compatibility, would be naturally before the file name, *and* the fact that writing `cases_dir` all the time is a bit awkward.

File: CHANGES.md
Comment (saroad2) [reply CHANGES.md:70]: The PR won't pass the CI process if this is missing.
If you want, and you will force-push the PR

Comment (JelleZijlstra) [reply CHANGES.md:70]: We can add a label to skip that check. I think it's fine to keep the news entry though, so we give credit to your contribution.

**Review by cooperlees — APPROVED** [review]
Just want to share my thanks, I am a HUGE fan of removing static manual lists like this!

#techdebtCleanup

**Comment by saroad2** [comment]
Hey everyone, thanks for the review

@ichard26's suggested API will work just fine at the moment without any change to code:

 ```python
 source, expected = read_data("simple_cases/expression.py")
 ```

However, I like @felix-hilden suggestion more since in the end there will be no cases in the top directory, but all should be placed in subdirectories according to their use-case:

```python
def read(subdir: str, file: str, ...):
    data_file = (TOP / subdir / file).with_suffix(".py")
```

My suggestion is this:
1. In this PR I should use the API suggested by @ichard26.
2. In the next PR I will move all cases to subdirectories and change the API to @felix-hilden suggestion.

Everyone agrees?

Two more things:
* Please let me know how to pass the CHANGES.md file check in order for me to remove this PR from the changelog. I agree with @ichard26 that since this is an internal change that is not user-facing it doesn't belong there.
*  @cooperlees , thanks for the complement! cleaning code and tests is one of my favorite things 😊

**Comment by felix-hilden** [comment]
That works! And we make non-changelog changes happen using a label which I applied just now 👍

**Comment by saroad2** [comment]
Great! I made all the changes we've discussed.

@felix-hilden , I really don't need that entry in the CHANGES.md to feel that I get credited for my work.
In my other PR I added myself to the AUTHORS.md, and that's all the credit I need for now.

*Black* is a tool that's really close to my heart and I'm happy to contribute for it whenever I have the time.

**Review by felix-hilden — APPROVED** [review]

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**
